lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -179,7 +181,7 @@ def initialize(obj)
       if obj.kind_of?(ActiveRecord::Base) && !obj.kind_of?(ar_klass)
         raise ArgumentError.new("#{ar_klass.name} Object expected, but received #{obj.class.name}")
       end
-      @object = obj.kind_of?(ar_klass) ? obj : ar_method { ar_klass.find_by_id(obj.to_i) }
+      @object = obj.kind_of?(ar_klass) ? obj : ar_method { self.class.find_ar_object_by_id(obj.to_i) }
```

Comment:
@lfu 
The initializer can get passed in an object or an id, we do the filtering only if we have an id. The RBAC search wouldn't work for objects that have been deleted from DB and are only available in memory.